capnp/src/io.rs

```diff
@@ -42,7 +42,7 @@ pub trait BufRead : Read {
 
 /// A rough approximation of std::io::Write.
 pub trait Write {
-    fn write_all(&mut self, buf: &[u8]) -> Result<()>;
+    fn write_all(&mut self, buf: &[u8]) -> Result<usize>;
```

Comment:
The `usize` return value does not give us any new information. An `Ok` result will always have written exactly `buf.len()` bytes.

Reply:
If I am not mistaken, it is not the case when using packed serialization.